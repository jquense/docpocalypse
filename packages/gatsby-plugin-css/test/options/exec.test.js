const { webpack } = require('../helpers');

xdescribe('Options', () => {
  test('Exec - {Boolean}', () => {
    const config = {
      loader: {
        test: /style\.(exec\.js|js)$/,
        options: {
          exec: true,
        },
      },
    };

    return webpack('jss/exec/index.js', config).then((stats) => {
      const { source } = stats.toJson().modules[1];

      expect(source).toEqual('module.exports = "a {\\n    color: green\\n}"');

      expect(source).toMatchSnapshot();
    });
  });

  test('JSS - {String}', () => {
    const config = {
      loader: {
        test: /style\.js$/,
        options: {
          parser: 'postcss-js',
        },
      },
    };

    return webpack('jss/index.js', config).then((stats) => {
      const { source } = stats.toJson().modules[1];

      expect(source).toMatchSnapshot();
    });
  });
});
