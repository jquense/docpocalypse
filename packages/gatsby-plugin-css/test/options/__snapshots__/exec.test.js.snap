// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options Exec - {Boolean} 1`] = `"module.exports = \\"a {\\\\n    color: green\\\\n}\\""`;

exports[`Options JSS - {String} 1`] = `"module.exports = \\"a {\\\\n    color: yellow\\\\n}\\""`;
